nixos/modules/services/databases/postgresql.nix

```diff
@@ -249,8 +285,12 @@ in
             else pkgs.postgresql_9_4);
 
     services.postgresql.dataDir =
-      mkDefault (if versionAtLeast config.system.stateVersion "17.09" then "/var/lib/postgresql/${config.services.postgresql.package.psqlSchema}"
-                 else "/var/db/postgresql");
+      mkDefault (if versionAtLeast config.system.stateVersion "17.09"
+                  then "/var/lib/postgresql/${cfg.package.psqlSchema}"
+                  else "/var/db/postgresql");
+
+    services.postgresql.initdbFlags =
+      mkDefault (lib.optional cfg.groupAccess "--allow-group-access");
```

Comment:
Setting both `groupAccess = true` and `initdbFlags = [ "--some-other-flag" ]` causes `"--allow-group"` not to be added due to `mkDefault`.

It's simpler and more explicit to combine arguments when calling `initdb` in `script` than making the latter option depend on the former.

Reply:
Indeed, thanks! I really shoud use `mkBefore` instead:
```
# nix repl ./lib
nix-repl> (evalModules { modules = [ 
                   { options.bla = mkOption { type = with types; listOf int; }; 
                     config.bla = mkBefore [1]; 
                   } 
                   { bla = [2];  } 
               ]; }).config.bla
[ 1 2 ]
```
Will change test.